BUILDING.md

```diff
@@ -1,56 +1,294 @@
 
-#### Additional information on dependencies
-* `libglm` (headers only) for matrix math is required. lives in `vendor/glm`.
-* A modified version of `hlslparser` is included in Renderer and used to transpile HLSL shaders to GLSL
-* OpenGL 3+ or OpenGLES is required
-* `libsdl >= 2.0.5` is required for the SDL and emscripten apps. `src/projectM-sdl` is the current reference application implementation. maybe try getting that to build and run as your testbench.
+#### Only relevant for Linux distributions, FreeBSD and macOS:
 
-If extra information needed - you can refere to `configure.ac` and the assorted `Makefile.am` files.
+* **pkgconfig**: Required to find some library dependencies.
+* [**Qt5**](https://www.qt.io/): Qt cross-platform UI framework. Used to build the Pulseaudio and JACK visualizer
+  applications. Requires the `Gui` and `OpenGL` component libraries/frameworks.
+* [**Pulseaudio**](https://www.freedesktop.org/wiki/Software/PulseAudio/): Sound system for POSIX platforms. Required to
+  build the Pulseaudio visualizer application.
+* [**JACK**](https://jackaudio.org/): Real-time audio server. Required to build the JACK visualizer application.
+* [**libvisual 0.4**](http://libvisual.org/): Audio visualization library with plug-in support. Required to build the
+  projectM libvisual plug-in.
 
-### Building process under *nix systems
-```sh
+#### When using the classic autotools-based build system on UNIX platforms:
+
+* **GNU [automake](https://www.gnu.org/software/automake/) and [autoconf](https://www.gnu.org/software/autoconf/)**:
+  Used to create the configuration script and generate the Makefiles.
+* [**libtool**](https://www.gnu.org/software/libtool/): Optional. Used by autoconf is available.
+* [**which**](https://carlowood.github.io/which/): Determines full paths of shell executables. Should already be
+  installed by default on the majority of POSIX-compliant OSes.
+
+#### When using the new CMake-based build system:
+
+* [**CMake**](https://cmake.org/): Used to generate platform-specific build files.
+
+### Only relevant for Windows:
+
+* [**vcpkg**](https://github.com/microsoft/vcpkg): C++ Library Manager for Windows. Optional, but recommended to install
+  the aforementioned library dependencies and/or using CMake to configure the build.
+* [**NuGet**](https://www.nuget.org/): Dependency manager for .NET. Optional, but recommended when building with the
+  pre-created Visual Studio solutions.
+* [**GLEW**](http://glew.sourceforge.net/): The OpenGL Extension Wrangler Library. Only required if using CMake to
+  configure the build, the pre-created solutions use a bundled copy of GLEW.
+
+## Building on Linux and macOS
+
+### Installing dependencies
+
+- Linux distributions will have packages available for most (if not all) required libraries. The package names and
+  commands to install them vary widely between distributions (and even versions of the same distribution). Please refer
```

Comment:
We had the list of debian packages above, would be nice to keep that since most people use that or know how to "translate" them to their weird pervert package managers

Reply:
Would then add a list of suggested packages for some distros, Homebrew and vcpkg, with the latter two also being an option on Linux as well.

Reply:
I think your documentation is swell and terrific and informative, just that 90% of users are just going to want to know "wtf do I type in to make projectM run" so there should be some really short quickstart section